wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsnext/model/CheEditor.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.workspace.server.wsnext.model;
+
+import java.util.Objects;
+
+/**
+ * Represents an editor inside of Che workspace.
+ *
+ * <p>It may be classic GWT IDE, Eclipse Theia or something else.
+ */
+public class CheEditor extends PluginBase {
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof CheEditor)) {
+      return false;
+    }
+    CheEditor cheEditor = (CheEditor) o;
+    return Objects.equals(getName(), cheEditor.getName())
```

Comment:
I think it would be better to write `equals` and `hashcode` methods in the same manner https://github.com/eclipse/che/pull/10457/commits/e89a9fb685af2823c802f3bfd53aae033f928533#diff-b3ca1f169efa0ad30c481db6568521a7R41

Reply:
Can you elaborate why it would be better?

Reply:
Let me clarify that I wanted to say that it is better to write these methods in the same way, but both ways are OK for me (with reusing super methods, or with duplicated piece of code).
I see two points here:
1. Nobody will ask why two very similar classes have a different style of `equals` and `hashcode` method.
2. One style is a less error-prone way because you get used to writing in this way, and you'll find an error easier.

like here https://github.com/eclipse/che/pull/10457/files#diff-a8707ef642f5ec96204cccfd4e370ce3R40 is not an error but overhead when `ChePluginBase` fields are used twice in hash code, the first time in `super.hashCode`, and the second time in CheEditor.

Reply:
I see. But you may also consider using super as a less error-prone approach since it reflects all the changes in a superclass automatically and that adding a field to a superclass and forgetting modifying a subclass doesn't introduce a bug.
I fixed the hashcode method, thanks for noticing it.